Messaging.md

```diff
@@ -165,10 +175,84 @@ This type contains signatures of the funding transaction and any necessary infor
 `witness` is the data for a witness element in a witness stack. An empty `witness_stack` is an error,
 as every input must be Segwit. Witness elements should *not* include their length as part of the witness data.
 
+### The `event_descriptor` Type
+
+This type contains information about the *exact* and fully specified outcomes  in an event for which an oracle plans on releasing a signature over.
+
+#### Version 0 `external_event_descriptor`
+
+1. type: 55300 (`external_event_descriptor_v0`)
+2. data:
+   * [`string`:`external_name`]
+
+`external_name` can refer to anything here and it is up to the oracle and user to agree on how to interpret it.
+
+#### Version 0 `enum_event_descriptor`
+
+1. type: 55302 (`enum_event_descriptor_v0`)
+2. data:
+   * [`u16`:`num_outcomes`]
+   * [`u16`:`outcome_1_len`]
+   * [`string`:`outcome_1`]
+   * ...
+   * [`u16`:`outcome_n_len`]
+   * [`string`:`outcome_n`]
+
+This type of event descriptor is a simple enumeration where the value `n` is the number of outcomes in the event.
+
+Each `outcome_i` corresponds to the pre-image of a possible outcome that the oracle could sign.
+
+#### Version 0 `range_event_descriptor`
+
+1. type: 55304 (`range_event_descriptor_v0`)
+2. data:
+   * [`int32`:`start`]
+   * [`int32`:`stop`]
+   * [`u16`:`step`]
+
+`start` refers to the first possible outcome number
+
+`end` refers to the last possible outcome number
+
+`step` refers to the increment between each outcome
+
+### The `oracle_event` Type
+
+This type contains information about an event for which an oracle plans on releasing a signature over.
+
+#### Version 0 `oracle_event`
+
+1. type: 55330 (`oracle_event_v0`)
+2. data:
+   * [`x_point`:`oracle_public_key`]
```

Comment:
I think public key should not be included in the TLV. You should verify the signature on the data before decoding it which by definition means you already know the public key. This can cause a bug where someone verifies the signature the public key included rather than the one they have stored.

Reply:
If the public key is not included how do you know which oracle to check against?

Reply:
Good point. I think a string `oracle_id` along with with the TLV where you send the signature. It is not signed but lets the client know which key to use (and if they don't have it, potentially where to get it). 

e.g.
- theoracle.com
- twitter.com/OutcomeObserver

Reply:
Another thing you could do for now is just move the public key into the `oracle_announcement` section which makes it a bit less awkward.